### PR TITLE
[GH-15580] Add information on UniformRobust method for `histogram_type` and create an accompanying blog post [nocheck]

### PR DESCRIPTION
**Pull request by hannah-tillman**
For: #15580 

This PR adds the UniformRobust method to the `histogram_type` algo-param page. It also drafts the initial blog post to be added to the User Guide blog section (I will add to the blog to the h2o.ai/blog/ website after the next release so the information in the Guide is up-to-date). 

The example I made for the blog might not be the best. I tried to find a dataset with decent outliers since I didn't have the Springleaf dataset mentioned in the original PR. Please let me know if it's okay or how I should update it.

### CODE REVIEW

File: h2o-docs/src/product/blog-releases.rst [hunk 1]
```diff
@@ -577,3 +577,88 @@ Prior Release Blogs
...
 
 You can find all prior release blogs `here <https://h2o.ai/blog/category/h2o-release/>`__.
+
+General Blogs
+-------------
+
+A Look at the UniformRobust method for ``histogram_type``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We introduced the `UniformRobust method for histogram <data-science/algo-params/histogram_type.html>`__ (``histogram_type="UniformRobust"``) in 3.36.1. and want to highlight its benefits! This method addresses issues found in the default UniformAdaptive method and preserve the runtime performance advantages compared to the QuantilesGlobal method.
```
Comment (meganjkurka) [inline h2o-docs/src/product/blog-releases.rst:587]: I think this uses our own terminology (QuantilesGlobal, UniformRobust) so this message will be missed on non-H2O employees or power users. 

The gist is that we have a new parameter that can increase accuracy for datasets with a lot of outliers without causing significant increase in time.

**Review by meganjkurka — COMMENTED** [review]
Hey I left some comments.

File: h2o-docs/src/product/blog-releases.rst [hunk 2]
```diff
@@ -577,3 +577,88 @@ Prior Release Blogs
...
+
+The introduction of the UniformRobust method mitigates these issues! By learning from histograms from the previous layer, we are able to fine-tune the split points for the current layer.
+
+To begin, the root of the tree uses UniformAdaptive binning. After the first split is made, we learn how the histograms were populated. Though this split has been made, the data has not yet been seen, so we do not yet know the exact distributions of the data in the right and left child nodes. If we see that UniformAdaptive binning fails, we can assume an issue still exists in the child nodes. Then, we approximate the distribution in the child nodes by the distribution in the parent node. We do this by looking at the bins of the histogram to check how many bins were empty compared to the total number of available bins. If too few bins are populated, that is when we switch to the new binning method.
```
Comment (meganjkurka) [inline h2o-docs/src/product/blog-releases.rst:593]: What does it mean for Uniform Adaptive binning to fail?

File: h2o-docs/src/product/blog-releases.rst [hunk 3]
```diff
@@ -577,3 +577,88 @@ Prior Release Blogs
...
+
+We introduced the `UniformRobust method for histogram <data-science/algo-params/histogram_type.html>`__ (``histogram_type="UniformRobust"``) in 3.36.1. and want to highlight its benefits! This method addresses issues found in the default UniformAdaptive method and preserve the runtime performance advantages compared to the QuantilesGlobal method.
+
+The issue found with the default UniformAdaptive method deals with outlier issues related to binning. This issue can be seen in how the UniformAdaptive method handles certain datasets with many columns that are primarily comprised of small integers where there are also columns with artificial values upwards of 99999, 99998. While you can certainly handle this issue in pre-processing steps, the default binning method can fail on features with certain characteristics. Uniform binning looks at the min and max values in the column from the observations in the current tree node. It splits the interval into bins of the same length. However, when there are such large outliers, these outliers are classified into a single bin while the rest of the observations end up in another single bin. This, unfortunately, leaves most of the bins unused.
```
[the remark this answers is not in the document]
Comment (hannah-tillman) [reply h2o-docs/src/product/blog-releases.rst:589]: These are great! Thank you :D

File: h2o-docs/src/product/blog-releases.rst (hunk 2 above)
Comment (hannah-tillman) [reply h2o-docs/src/product/blog-releases.rst:593]: My guess is along the lines of "fails to distribute/bin evenly"? But I am just going off what is available in the [original PR](https://github.com/h2oai/h2o-3/pull/6063). 

@maurever any guidance?

File: h2o-docs/src/product/blog-releases.rst [hunk 4]
```diff
@@ -577,3 +577,96 @@ Prior Release Blogs
...
+
+For the next level of split-points, we then take the boundaries of the non-empty bins and refine them according to the squared error that is accumulated in each bin. Non-empty bins with a higher squared error are split more than ones with a lower squared error. This act of splitting creates sub-bins, and these are refined uniformly. We aid this process by freeing up space by discarding empty bins based on the desired target number of bins.
+
+So, if uniform splitting fails, the next iteration attempts to correct the issue by repeating the procedure with new bins. This allows us to recursively refine the promising bins as we get deeper in the tree.
```
Comment (maurever) [inline h2o-docs/src/product/blog-releases.rst:601]:
```suggestion
So, if the case the uniformRobust splitting fails (the distribution of values is still significantly skewed), the next iteration of finding splits attempts to correct the issue by repeating the procedure with new bins. This allows us to refine the promising bins recursively as we get deeper into the tree.
```

File: h2o-docs/src/product/data-science/algo-params/histogram_type.rst [hunk 5]
```diff
@@ -19,12 +19,15 @@ By default (AUTO) GBM/DRF bins from min...max in steps of (max-min)/N.  Use this
...
 - AUTO
 - UniformAdaptive
+- UniformRobust
 - Random
 - QuantilesGlobal
 - RoundRobin
 
 When ``histogram_type="UniformAdaptive"`` is specified, each feature is binned into buckets of equal step size (not population). This is the fastest method, and usually performs well, but can lead to less accurate splits if the distribution is highly skewed.
 
+When ``histogram_type="UniformRobust"`` is specified, uniform binning is used at the root of the tree. If the dataset has large outliers, this method can lead to poor distribution. When this happens, it takes the boundaries of the non-empty bins and refines them according to the squared error accumulated in each bin. Non-empty bins with higher squared error are split more than ones with lower squared error to create sub-bins which are refined uniformly. So, if uniform splitting fails, the next iteration attempts to correct the issue by repeating the procedure with new bins. This allows it to recursively refine the promising bins as it get deeper in the tree. This method is highly effective on datasets with large outliers.
```
Comment (maurever) [inline h2o-docs/src/product/data-science/algo-params/histogram_type.rst:29]:
```suggestion
If the dataset has outliers, the uniform method can lead to poor distribution. When ``histogram_type="UniformRobust"`` is specified, uniform binning is used for finding initial splits, then the bins are redefined the data will be better distributed over empty bins. It takes the boundaries of the non-empty bins and refines them according to the squared error accumulated in each bin. Non-empty bins with higher squared error are split more than ones with lower squared error to create sub-bins that are refined uniformly. So, if uniform splitting fails, the next iteration of finding splits attempts to correct the issue by repeating the procedure with new bins. This allows it to recursively refine the promising bins as it gets deeper into the tree. This method is highly effective on datasets with large outliers.
```

**Review by maurever — CHANGES_REQUESTED** [review]
@hannah-tillman, thanks for your work. I reformat some sentences to be more precious and explainable. Otherwise, it looks good to me. Please check the grammar is ok before implementing the suggested changes. :) Please don't hesitate to let me know if you have any other questions.

File: h2o-docs/src/product/blog-releases.rst (hunk 2 above)
Comment (hannah-tillman) [reply h2o-docs/src/product/blog-releases.rst]: I wasn't sure how in-depth to go with this explanation, so I did start with the minimum:
`If we see that UniformAdaptive binning fails (i.e. there are a lot of empty bins)...` 
but please let me know if I should add more.

**Review by maurever — APPROVED** [review]
@hannah-tillman, thanks! Looks good to me. :)

**Pull request merged**